packages/firestore/src/local/shared_client_state_syncer.ts

```diff
@@ -14,10 +14,26 @@
  * limitations under the License.
  */
 
-import { BatchId, MutationBatchState } from '../core/types';
+import { BatchId, MutationBatchState, TargetId } from '../core/types';
 import { FirestoreError } from '../util/error';
 import { ClientId } from './shared_client_state';
 
+/**
+ * The different states of a watch target.
+ *
+ * When a secondary tab inserts a new target, it is marked 'pending'. The
+ * primary tab then transitions the target to 'incomplete' once Watch has
+ * delivered the first snapshot. Once a Watch target is current, it transitions
+ * to 'current'.
+ *
+ * Targets that get rejected from the backend are marked 'rejected'.
+ */
+export type QueryTargetState =
+  | 'pending'
```

Comment:
It's not obvious to me why we need this state.  What will secondary tabs do in response to 'pending'?  If we got rid of it then we could use QueryTargetState everywhere, right?

Reply:
This was a way to avoid another callback. I have added the callback with the name we discussed on Friday (applyActiveTargetsChange).